src/marina_pool_server.erl

```diff
@@ -131,30 +136,35 @@ connect(Ip, Port) ->
             {error, Reason}
     end.
 
-filter_datacenter([], _Datacenter) ->
+filter_datacenter_and_racks([], _Datacenter, _Configured_Racks) ->
     [];
-filter_datacenter([[RpcAddress, _Datacenter, Tokens] | T], undefined) ->
-    [{RpcAddress, Tokens} | filter_datacenter(T, undefined)];
-filter_datacenter([[RpcAddress, Datacenter, Tokens] | T], Datacenter) ->
-    [{RpcAddress, Tokens} | filter_datacenter(T, Datacenter)];
-filter_datacenter([_ | T], Datacenter) ->
-    filter_datacenter(T, Datacenter).
-
-nodes([], _Port) ->
+filter_datacenter_and_racks([[RpcAddress, Datacenter, Rack, Tokens] | T], Retrieved_Datacenter, Configured_Racks)
+    when Retrieved_Datacenter == undefied; Retrieved_Datacenter == Datacenter->
```

Comment:
Also, what's the use case for undefined data center?

Reply:
undefined data center is where there no cluster. e.g. - localhost and one node.